File: docker/Dockerfile.java
FROM openjdk:17-slim

WORKDIR /app

# Instalar dependencias del sistema
RUN apt-get update && apt-get install -y \
    curl \
    && rm -rf /var/lib/apt/lists/*

# Copiar código fuente Java
COPY src/java/ .

# Compilar aplicación Java
RUN javac *.java

# Crear directorio para descargas
RUN mkdir -p downloads

# Comando por defecto
CMD ["java", "com.filedownloader.Main"]

File: docker/Dockerfile.go
FROM golang:1.21-alpine AS builder

WORKDIR /app

# Copiar archivos Go
COPY src/go/ .

# Compilar aplicación
RUN go build -o filedownloader filedownloader.go

# Imagen final
FROM alpine:latest

WORKDIR /app

# Instalar certificados SSL
RUN apk --no-cache add ca-certificates

# Copiar ejecutable
COPY --from=builder /app/filedownloader .

# Crear directorio para descargas
RUN mkdir -p downloads

# Comando por defecto
CMD ["./filedownloader"]